src/Methods/InvoicePaymentMethod.php

```diff
@@ -67,86 +71,36 @@ public function __construct(
-                   if($method instanceof ContactAllowedMethodOfPayment) {
-                       if($method->methodOfPaymentId == $this->invoiceHelper->getInvoiceMopId() && $method->allowed) {
-                           return true;
-                       }
-                   }
-                });
-                if($allowed) {
-                    return true;
-                }
-
-                if((int)$this->settings->getSetting('quorumOrders') > 0 && $contact->orderSummary->orderCount < $this->settings->getSetting('quorumOrders')) {
-                    return false;
-                }
-            }
-        } elseif ((int)$this->settings->getSetting('quorumOrders') > 0 && !$this->accountService->getIsAccountLoggedIn()) {
-            return false;
-        }
         
-        /** @var CurrencyExchangeRepository $currencyService */
-        $currencyService = pluginApp(CurrencyExchangeRepositoryContract::class);
-        $minAmount = (float)$currencyService->convertFromDefaultCurrency($basket->currency, $this->settings->getSetting('minimumAmount'));
-        $maxAmount = (float)$currencyService->convertFromDefaultCurrency($basket->currency, $this->settings->getSetting('maximumAmount'));
-
-        /**
-         * Check the minimum amount
-         */
-        if( $minAmount > 0.00 && $basket->basketAmount < $minAmount)
-        {
-            return false;
-        }
-
-        /**
-         * Check the maximum amount
-         */
-        if( $maxAmount > 0.00 && $maxAmount < $basket->basketAmount)
-        {
-            return false;
-        }
-
-
-        /**
-         * Check whether the invoice address is the same as the shipping address
-         */
-        if( $this->settings->getSetting('invoiceEqualsShippingAddress',$lang) == 1)
-        {
-            $invoiceAddressId = $basket->customerInvoiceAddressId;
-            $shippingAddressId = $basket->customerShippingAddressId;
-
-            if($shippingAddressId != null && $invoiceAddressId != $shippingAddressId)
-            {
-                return false;
-            }
-        }
-
-        /**
-         * Check whether the user is logged in
-         */
-        if( $this->settings->getSetting('disallowInvoiceForGuest',$lang) == 1 && !$this->accountService->getIsAccountLoggedIn())
-        {
-            return false;
-        }
-
-        if(!in_array($this->checkout->getShippingCountryId(), $this->settings->getShippingCountries()))
-        {
-            return false;
+        $isGuest = !($this->accountService->getIsAccountLoggedIn() && $basket->customerId > 0);
+        $contact = null;
+        if(!$isGuest && $basket->customerId > 0) {
+            try {
+                $contact = $contactRepo->findContactById($basket->customerId, ['orderSummary']);
+            } catch(\Exception $ex) {}
         }
-
-        return true;
+        $amount = $amount = pluginApp(OrderAmount::class);
```

Comment:
I think this may have been a copy & paste thing here? Or why did you double the allocation of the variable `$amount`?